library/manageiq_vmdb.py

```diff
@@ -92,13 +92,17 @@ def build_result(self, method, data=None):
         """
         result, info = fetch_url(self._module, self.url, data, self._headers, method)
         try:
+            if info["status"] >= 400:
+                error = json.loads(info["body"])
+                self._module.fail_json(msg=error)
+
             vmdb = json.loads(result.read())
             if self._debug:
                 vmdb['debug'] = info
+
             return vmdb
-        except AttributeError:
+        except (json.decoder.JSONDecodeError, AttributeError):
             self._module.fail_json(msg=info)
-        return json.loads(result.read())
```

Comment:
what was this returning previously?  Was it trying it again?

Reply:
So, in the case of an error, the result result.read would be empty.  The error content is stored in the `info["body"]`.

Reply:
Oh, wait, this particular line was unreachable - seems it was an oversight

Reply:
Thanks, I was confused.